src/main/java/net/oauth/jsontoken/discovery/AsyncVerifierProvider.java

```diff
@@ -36,14 +37,15 @@
 public interface AsyncVerifierProvider {
 
   /**
-   * Returns a {@link ListenableFuture}, which asynchronously returns a {@link List<Verifier>}
+   * Returns a {@link ListenableFuture}, which asynchronously returns a {@code List<Verifier>}
    * that represents a certain verification key, given the key's id and its issuer.
    * @param issuer the id of the issuer that's using the key.
    * @param keyId the id of the key, if keyId mismatches, return a list of
    *   possible verification keys.
-   * @return a {@link ListenableFuture} object that asynchronously returns a {@link List<Verifier>}
+   * @return a {@link ListenableFuture} object that asynchronously returns a {@code List<Verifier>}
    * that represents the verification key.
    */
+  @Nonnull
```

Comment:
This only enforces the returned ListenableFuture to be nonnull, not the wrapped List of verifiers right?

Reply:
Yes, the nonnull annotation is intended to be on the returned ListenableFuture.